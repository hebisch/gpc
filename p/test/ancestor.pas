{ FLAG -Dwallclock=42 }

program ancestor(output);
{R.W.Floyd: 'Ancestor', Comm.ACM 6-62 and 3-63, Alg.96}
  const n = 20;
  var i,j,k: integer;
      r: array [1..n, 1..n] of boolean;
begin { r[i,j] = "i is a parent of j"}
  for i := 1 to n do
    for j := 1 to n do r[i,j] := false;
  for i := 1 to n do
    if i mod 10 <> 0 then r[i,i+1] := true;
  writeln(wallclock);
  for i := 1 to n do
    for j := 1 to n do
      if r[j,i] then
        for k := 1 to n do
          if r[i,k] then r[j,k] := true;
  writeln(wallclock);
  for i := 1 to n do
  begin write(' ');
    for j := 1 to n do write(chr(ord(r[i,j])+ord('0')));
    writeln
  end ;
  writeln(wallclock)
end .
