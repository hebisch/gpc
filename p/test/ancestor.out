42
42
 01111111110000000000
 00111111110000000000
 00011111110000000000
 00001111110000000000
 00000111110000000000
 00000011110000000000
 00000001110000000000
 00000000110000000000
 00000000010000000000
 00000000000000000000
 00000000000111111111
 00000000000011111111
 00000000000001111111
 00000000000000111111
 00000000000000011111
 00000000000000001111
 00000000000000000111
 00000000000000000011
 00000000000000000001
 00000000000000000000
42
